src_c/rwobject.c

```diff
@@ -243,8 +243,18 @@ pg_EncodeString(PyObject *obj, const char *encoding, const char *errors,
 static PyObject *
 pg_EncodeFilePath(PyObject *obj, PyObject *eclass)
 {
+#if PY_VERSION_HEX >= 0x030c0000
+    PyConfig config;
+    _PyInterpreterState_GetConfigCopy(&config);
```

Comment:
Should this be `PyConfig_InitPythonConfig` or something? I don't think it's a good idea to use private python C API

Reply:
There's probably another way to do it, but @vstinner is on holiday and he's my go-to info source on IRC for c-api :)
iirc his py init rework was after 3.7b.

Reply:
Are you looking for PyUnicode_EncodeFSDefault()? Calling _PyInterpreterState_GetConfigCopy() here sounds overkill. Casting config.filesystem_encoding wide-string (``wchar_t*`` type) to a byte string (``char*``) sounds plain wrong.

Reply:
indeed PyUnicode_EncodeFSDefault was my first pick, but i obviously don't know how to use it correctly as i made it segfault.

Reply:
Is this PR still good, or is it going to go back to `PyUnicode_EncodeFSDefault`?

Reply:
For me, PyUnicode_EncodeFSDefault() should be used. No idea why you got a crash, you didn't provide any detail (OS, Python version, error message, traceback, etc.).

Reply:
thanks for the advice, let's try again.

Reply:
it crashed , i'll add fencing for >=3.7 because it seems to be the last change here https://docs.python.org/3/c-api/unicode.html#c.PyUnicode_EncodeLocale

Reply:
crashes with  cp37-manylinux_x86_64

Reply:
I've mentioned the reason for this segfault (that I think) in my first comment, `PyUnicode_EncodeFSDefault` is simply not a good full replacement for our custom function. Callers expect `pg_EncodeString` to forward errors when it gets `NULL`, but `PyUnicode_EncodeFSDefault` is segfaulting trying to deref a null pointer (which makes sense because forwarding errors is not a documented API detail for this function, and we can't rely on this)

Fencing version-wise won't fix this segfault, we need to either fix the usage of this function in all instances, or add a null check at the start and forward it

Reply:
3.8 so i'm really using it wrong and i can't see where

Reply:
PyUnicode_EncodeFSDefault() must not be called on NULL. Why not adding a simple check to see if obj is NULL?

Reply:
good idea